src/Generators/TestsGenerator.php

```diff
@@ -68,7 +71,28 @@ protected function getInserts()
                     ]
                 ]
             ];
-        }, $this->buildRelationsTree($arrayModels));
+        }, $this->getRelationsWithFactories($relations));
+    }
+
+    protected function getRelationsWithFactories($relations)
+    {
+        $factory = app(Factory::class);
+
+        $relationsWithFactories = [];
+
+        foreach ($relations as $relation) {
+            if (
+                !empty($factory[$this->getModelClass($relation)])
+                || (
+                    $this->classExists('factory', "{$relation}Factory")
+                    && method_exists($this->getModelClass($relation), 'factory')
+                )
```

Comment:
could you please simplify the condition? by using variables or protected methdods